consensus/polybft/bridge_event_manager.go

```diff
@@ -0,0 +1,570 @@
+	log *ethgo.Log, dbTx *bolt.Tx) error {
+	return nil
+}
+
+// bridgeEventManagerConfig holds the configuration data of bridge event manager
+type bridgeEventManagerConfig struct {
+	topic             topic
+	key               *wallet.Key
+	maxNumberOfEvents uint64
+}
+
+var _ BridgeEventManager = (*bridgeEventManager)(nil)
+
+// bridgeEventManager is a struct that manages the workflow of
+// saving and querying bridge message events, and creating, and submitting new batches
+type bridgeEventManager struct {
+	logger hclog.Logger
+	state  *State
+
+	config *bridgeEventManagerConfig
+
+	// per epoch fields
+	lock                   sync.RWMutex
+	pendingBridgeBatches   []*PendingBridgeBatch
+	validatorSet           validator.ValidatorSet
+	epoch                  uint64
+	nextBridgeEventIDIndex uint64
+	sourceChainID          uint64
+
+	runtime Runtime
+}
+
+// topic is an interface for p2p message gossiping
+type topic interface {
+	Publish(obj proto.Message) error
+	Subscribe(handler func(obj interface{}, from peer.ID)) error
+}
+
+// newBridgeEventManager creates a new instance of bridge event manager
+func newBridgeEventManager(logger hclog.Logger, state *State, config *bridgeEventManagerConfig,
+	runtime Runtime, sourceChainID uint64) *bridgeEventManager {
+	return &bridgeEventManager{
+		logger:        logger,
+		state:         state,
+		config:        config,
+		runtime:       runtime,
+		sourceChainID: sourceChainID,
+	}
+}
+
+// Init subscribes to bridge topics (getting votes)
+func (b *bridgeEventManager) Init() error {
+	if err := b.initTransport(); err != nil {
+		return fmt.Errorf("failed to initialize bridge event transport layer. Error: %w", err)
+	}
+
+	return nil
+}
+
+// initTransport subscribes to bridge topics (getting votes for batches)
+func (b *bridgeEventManager) initTransport() error {
+	return b.config.topic.Subscribe(func(obj interface{}, _ peer.ID) {
+		if !b.runtime.IsActiveValidator() {
+			// don't save votes if not a validator
+			return
+		}
+
+		msg, ok := obj.(*polybftProto.TransportMessage)
+		if !ok {
+			b.logger.Warn("failed to deliver vote, invalid msg", "obj", obj)
+
+			return
+		}
+
+		var transportMsg *TransportMessage
+
+		if err := json.Unmarshal(msg.Data, &transportMsg); err != nil {
+			b.logger.Warn("failed to deliver vote", "error", err)
+
+			return
+		}
+
+		if err := b.saveVote(transportMsg); err != nil {
+			b.logger.Warn("failed to deliver vote", "error", err)
+		}
+	})
+}
+
+// saveVote saves the gotten vote to boltDb for later quorum check and signature aggregation
+func (b *bridgeEventManager) saveVote(msg *TransportMessage) error {
+	b.lock.RLock()
+	epoch := b.epoch
+	valSet := b.validatorSet
+	b.lock.RUnlock()
+
+	if valSet == nil || msg.EpochNumber < epoch || msg.EpochNumber > epoch+1 || b.sourceChainID != msg.SourceChainID {
+		// Epoch metadata is undefined or received a message for the irrelevant epoch
+		return nil
+	}
```

Comment:
Maybe it would be beneficial to add debug logs at some point and check all these conditions separately? WDYT?